react-components/src/components/CadModelContainer/CadModelContainer.tsx

```diff
@@ -27,29 +55,67 @@ export function CadModelContainer({
   }, [modelId, revisionId, geometryFilter]);
 
   useEffect(() => {
-    if (modelRef.current === undefined || transform === undefined) return;
-    modelRef.current.setModelTransformation(transform);
-  }, [transform]);
+    if (model === undefined || transform === undefined) return;
+    model.setModelTransformation(transform);
+  }, [transform, model]);
+
+  useEffect(() => {
+    if (model === undefined || styling === undefined) return;
+
+    applyStyling(sdk, model, styling).catch(console.error);
+
+    return () => {
+      model.removeAllStyledNodeCollections();
+      model.setDefaultNodeAppearance(DefaultNodeAppearance.Default);
+    };
+  }, [styling, model]);
 
   return <></>;
 
   async function addModel(
     modelId: number,
     revisionId: number,
     transform?: Matrix4,
-    onLoad?: () => void
-  ): Promise<void> {
+    onLoad?: (model: CogniteCadModel) => void
+  ): Promise<CogniteCadModel> {
```

Comment:
I don't think it is valuable to expose the model when current use case does not need it, when needed in future we can expose it

Reply:
I still think that it is very reasonable to add the model there because you would expect something as an argument of `onLoad` function and if you want any custom functionality you have to access the model somehow anyway.

Reply:
I am good with it for now!